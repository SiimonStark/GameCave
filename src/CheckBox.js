import React from 'react';

export default function CheckBox (props) {
	return (
		<div className="genre-container">
			<label htmlFor={props.name}>
				<input type="checkbox" className={props.class} id={props.name} onChange= {props.updateChecks}/>
				<div className="faux-box"></div>
				<p className="genre-title">{props.name}</p>
			</label>
		</div>
	);
}