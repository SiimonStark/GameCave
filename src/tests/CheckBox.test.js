import React from 'react';
import ReactDOM from 'react-dom';
import CheckBox from '../CheckBox';
import { shallow } from 'enzyme';

const updateChecksMock = jest.fn();

describe('CheckBox', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CheckBox 
        updateChecks={updateChecksMock}
      />
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateChecks on app on input change', () => {
    wrapper.find('input').simulate('change');
    expect(updateChecksMock).toBeCalled();
  })
})