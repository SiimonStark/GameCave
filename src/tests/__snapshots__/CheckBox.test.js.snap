// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should match snapshop 1`] = `
<div
  className="genre-container"
>
  <label>
    <input
      onChange={[MockFunction]}
      type="checkbox"
    />
    <div
      className="faux-box"
    />
    <p
      className="genre-title"
    />
  </label>
</div>
`;
